lib/parser/provenance.cc

```diff
@@ -241,6 +253,26 @@ const SourceFile *AllSources::GetSourceFile(
       origin.u);
 }
 
+std::optional<std::pair<int, int>> AllSources::GetSourcePosition(
+    Provenance prov) const {
+  const Origin &origin{MapToOrigin(prov)};
+  if (auto inc = std::get_if<Inclusion>(&origin.u)) {
+    std::size_t offset{origin.covers.MemberOffset(prov)};
+    return {std::pair<int, int>(inc->source.FindOffsetLineAndColumn(offset))};
```

Comment:
You probably don't need the braces or the `pair` constructor; `FindOffsetLineAndColumn()` returns a `pair`, yes?